Extensions/xkit_patches.js

```diff
@@ -664,10 +664,17 @@ XKit.tools.dump_config = function(){
 					if ($(".html-field").css("display") === "none") {
 						var content_editor = $('.post-form--form').find('.editor.editor-richtext');
 						if (content_editor.length === 0) {
-							XKit.console.add('ERROR: unable to set content html');
+							XKit.console.add('ERROR: unable to get content html');
 							return '';
 						}
 						return content_editor.html();
+					} else {
+						var html_or_markdown = $(".tab-label[data-js-srclabel]").text();
+						if (html_or_markdown === 'HTML') {
+							return $('.ace_editor .ace_text-layer').text();
+						}
+						XKit.console.add('ERROR: unable to get content html for markdown editor');
```

Comment:
To be specific, I know it's possible to make it work with Markdown, but I
haven't gotten the API call working yet.

On Thu, Sep 17, 2015, 09:54 BlackjackKent <notifications@github.com> wrote:

> In Extensions/xkit_patches.js
> <https://github.com/new-xkit/XKit/pull/504#discussion_r39769770>:
>
> >  							return '';
> >  						}
> >  						return content_editor.html();
> > +					} else {
> > +						var html_or_markdown = $(".tab-label[data-js-srclabel]").text();
> > +						if (html_or_markdown === 'HTML') {
> > +							return $('.ace_editor .ace_text-layer').text();
> > +						}
> > +						XKit.console.add('ERROR: unable to get content html for markdown editor');
>
> It succeeds if the editor is HTML and fails if it is Markdown. (@bvtsang
> <https://github.com/bvtsang> figured out a way to make this work with
> Markdown too but it's going to take extra work and I'm going to do it
> separately)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/new-xkit/XKit/pull/504/files#r39769770>.
>